llvm/lib/SYCL/PrepareSYCLOpt.cpp

```diff
@@ -230,12 +231,55 @@ struct PrepareSYCLOptState {
     }
   }
 
+  /// transform call on casted functions into call on functions with casted
+  /// arguments. because calls on casted function are not inlined and the Vitis
+  /// backend has issues with call on casted functions
```

Comment:
```suggestion
  /// Transform calls on casted functions into calls on functions with casted
  /// arguments, because calls on casted functions are not inlined and the Vitis
  /// backend has issues with call on casted functions.
```